### PR TITLE
SUP-1503 Add support for step-level branch filtering

### PR DESCRIPTION
**Pull request by lizrabuya**
This PR adds step-level branch filtering for the [on.push ](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore) event workflow and supports `branches` and `branches-ignore`

### CODE REVIEW

File: app/lib/bk/compat/parsers/gha.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 require_relative '../translator'
 require_relative '../pipeline'
 require_relative 'gha/jobs'
-require_relative 'gha/steps'
+require_relative 'gha/steps'
```
Comment (mcncl) [inline app/lib/bk/compat/parsers/gha.rb:7]: Additional white space can be removed.

File: app/lib/bk/compat/parsers/gha.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ def self.option
       end
 
       def self.matches?(text)
-        config = YAML.safe_load(text)
+        config = YAML.safe_load(text)
```
Comment (mcncl) [inline app/lib/bk/compat/parsers/gha.rb:25]: Same here.

File: app/spec/lib/bk/compat/__snapshots__/examples/github-actions/basic.yaml.snap [hunk 3]
```diff
@@ -7,3 +7,4 @@ steps:
     runs-on: ubuntu-latest
   label: ":github: explore"
   key: explore
+  branches: ''
```
Comment (mcncl) [inline app/spec/lib/bk/compat/__snapshots__/examples/github-actions/basic.yaml.snap:10]: It doesn't seem to be handling null? Shouldn't we _not_ have the filter if one isn't set?

**Review by mcncl — CHANGES_REQUESTED** [review]

File: app/spec/lib/bk/compat/__snapshots__/examples/github-actions/basic.yaml.snap (hunk 3 above)
Comment (lizrabuya) [reply app/spec/lib/bk/compat/__snapshots__/examples/github-actions/basic.yaml.snap:10]: Good catch. I did try different scenarios and this has regressed in the last changes I made.

Comment (lizrabuya) [reply app/spec/lib/bk/compat/__snapshots__/examples/github-actions/basic.yaml.snap]: @mcncl , fixed in latest commit

**Review by mcncl — APPROVED** [review]

**Review by mcncl — APPROVED** [review]

**Pull request merged**